### PR TITLE
fix(service): make bullet deduplication company-aware (BUG-013)

### PR DESCRIPTION
**Pull request by baphled**
## Summary

- **BUG-013**: `deduplicateBullets()` used normalized text as the sole dedup key, merging bullets with identical descriptions across different companies into a single bullet. This caused `SourceEventIDs` to span multiple employers and corrupted downstream date range calculations (e.g., BEIS showing "Nov 2012 - Dec 2019" from a merged We Are Friday event).
- The dedup key is now a struct-based composite of `{text, company}`, resolved from source event IDs via a new `resolvePrimaryCompany()` helper. Identical text at the same company is still merged; identical text across companies is kept separate.
- Addressed Copilot review feedback: deterministic lexicographic tie-breaking in `resolvePrimaryCompany`, struct-based dedup key (avoids pipe delimiter collisions), fallback to first SourceEventID when company cannot be resolved, and bullet ID fallback for orphaned bullets with no SourceEventIDs.
- Filed **BUG-015** for missing Company/Project validation on `CareerEvent.Validate()`.
- **SQLite locking fix**: Centralised all SQLite connection setup in `OpenDB()` with WAL journal mode, 5-second busy timeout, and single-connection pool. Fixes `SQLITE_BUSY` errors in Windows CI E2E tests.
- Adds 7 regression tests covering: cross-company separation, same-company merging, cross-cutting entries across 5 companies, SourceEventID scoping validation, deterministic tie-breaking (direct unit test), empty-company fallback, and orphaned bullet dedup.

## Changes

| File | Change |
|------|--------|
| `internal/service/career/cv/bullet_generator.go` | Struct-based dedup key; deterministic `resolvePrimaryCompany()` with lexicographic tie-breaking; empty-company fallback to first SourceEventID; bullet ID fallback for orphaned bullets; event map built in `GenerateBullets()` |
| `internal/service/career/cv/bullet_generator_test.go` | 7 new BUG-013 regression tests |
| `internal/repository/career/sql/repositories.go` | New `OpenDB()` helper with WAL mode, busy_timeout, and `SetMaxOpenConns(1)` |
| `cmd/cli/main.go` | Use `careersql.OpenDB()` instead of bare `sql.Open()` |
| `internal/testutil/db.go` | Use `careersql.OpenDB()` for test DB setup |
| `internal/testutil/e2e/helpers.go` | Use `careersql.OpenDB()` in all 3 E2E setup functions |
| `bugs/BUG-013-bullet-dedup-cross-company.md` | Bug report documenting root cause and fix |
| `bugs/BUG-015-career-event-missing-association-validation.md` | Bug report for missing Company/Project validation |

## Testing

- All 363 tests in the CV package pass (including 7 new)
- All E2E tests pass with race detector
- `go vet` and `gofmt` clean
- SQLite locking fix verified locally; Windows CI previously failed with 31 SQLITE_BUSY errors

### CODE REVIEW

File: internal/service/career/cv/bullet_generator.go [hunk 1]
```diff
@@ -439,6 +451,32 @@ func (bg *DefaultBulletGenerator) deduplicateBullets(bullets []*Bullet) []*Bulle
...
+		}
+	}
+
+	primaryCompany := ""
+	maxCount := 0
+	for company, count := range companyCounts {
+		if count > maxCount {
+			maxCount = count
+			primaryCompany = company
+		}
```
[the remark this answers is not in the document]
Comment (baphled) [reply internal/service/career/cv/bullet_generator.go]: Fixed in b6e695c9. Company names are now sorted lexicographically before selection, so ties are broken deterministically.

File: internal/service/career/cv/bullet_generator.go [hunk 2]
```diff
@@ -398,38 +404,44 @@ func (bg *DefaultBulletGenerator) extractPrimaryCategory(categories []string) co
...
-		// Normalize text for comparison (lowercase, trim spaces)
+		// Normalize text for comparison (lowercase, trim spaces).
 		normalizedText := strings.ToLower(strings.TrimSpace(bullet.Text))
 
-		if existing, exists := seen[normalizedText]; exists {
-			// Merge: keep the one with higher confidence, combine source IDs
+		// Resolve primary company from source events (BUG-013).
+		company := resolvePrimaryCompany(bullet.SourceEventIDs, eventMap)
+		key := normalizedText + "|" + company
+
```
[the remark this answers is not in the document]
Comment (baphled) [reply internal/service/career/cv/bullet_generator.go]: Fixed in b6e695c9. When company cannot be resolved, we now fall back to the first SourceEventID to prevent unrelated bullets merging under an empty key. Also filed BUG-015 to add upstream validation requiring events to have either a Company or Project.

File: internal/service/career/cv/bullet_generator_test.go [hunk 3]
```diff
@@ -1149,3 +1150,86 @@ var _ = Describe("BUG-008: ScoringConfig integration", func() {
...
+		for _, bullet := range bullets {
+			companies := map[string]bool{}
+			for _, eid := range bullet.SourceEventIDs {
+				companies[eventCompany[eid]] = true
+			}
+			Expect(companies).To(HaveLen(1),
+				"SourceEventIDs should only reference events from one company, got %v", bullet.SourceEventIDs)
+		}
+	})
+})
```
[the remark this answers is not in the document]
Comment (baphled) [reply internal/service/career/cv/bullet_generator_test.go:1285]: This edge case should be handled via domain validation rather than defensive tests. An event should always have either a company or project association. Filed BUG-015 to add `validateAssociation()` to `CareerEvent.Validate()`, which will reject events with both fields empty at the domain boundary.

File: internal/service/career/cv/bullet_generator.go [hunk 4]
```diff
@@ -398,38 +404,44 @@ func (bg *DefaultBulletGenerator) extractPrimaryCategory(categories []string) co
...
 	for _, bullet := range bullets {
-		// Normalize text for comparison (lowercase, trim spaces)
+		// Normalize text for comparison (lowercase, trim spaces).
 		normalizedText := strings.ToLower(strings.TrimSpace(bullet.Text))
 
-		if existing, exists := seen[normalizedText]; exists {
-			// Merge: keep the one with higher confidence, combine source IDs
+		// Resolve primary company from source events (BUG-013).
+		company := resolvePrimaryCompany(bullet.SourceEventIDs, eventMap)
+		key := normalizedText + "|" + company
```
[the remark this answers is not in the document]
Comment (baphled) [reply internal/service/career/cv/bullet_generator.go]: Fixed in b6e695c9. Replaced the string concatenation key with a `dedupKey` struct, eliminating any delimiter collision risk.

File: internal/service/career/cv/bullet_generator_test.go [hunk 5]
```diff
@@ -1149,3 +1150,121 @@ var _ = Describe("BUG-008: ScoringConfig integration", func() {
...
+			bullets, err := generator.GenerateBullets(ctx, events, nil, nil, "", "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bullets).To(HaveLen(2), "different companies must produce separate bullets")
+			if i == 0 {
+				firstResult = bullets[0].Text
+			}
+			Expect(bullets[0].Text).To(Equal(firstResult),
+				"iteration %d: primary company resolution must be deterministic", i)
+		}
+	})
```
[the remark this answers is not in the document]
Comment (baphled) [reply internal/service/career/cv/bullet_generator_test.go:1255]: Fixed in 0aeb0d0e. Rewrote the test to call `resolvePrimaryCompany` directly with genuinely tied input (two source events from different companies with equal counts). Asserts `Alpha Corp` wins lexicographically and verifies determinism over 20 iterations.

File: internal/service/career/cv/bullet_generator.go [hunk 6]
```diff
@@ -398,38 +404,59 @@ func (bg *DefaultBulletGenerator) extractPrimaryCategory(categories []string) co
...
-			// Merge: keep the one with higher confidence, combine source IDs
+		// Resolve primary company from source events (BUG-013).
+		company := resolvePrimaryCompany(bullet.SourceEventIDs, eventMap)
+
+		// When company cannot be resolved (no company on source events),
+		// fall back to the first SourceEventID to prevent unrelated bullets
+		// from merging under an empty key (BUG-015 defence-in-depth).
+		if company == "" && len(bullet.SourceEventIDs) > 0 {
+			company = bullet.SourceEventIDs[0]
+		}
```
[the remark this answers is not in the document]
Comment (baphled) [reply internal/service/career/cv/bullet_generator.go]: Fixed in 0aeb0d0e. When both company is empty and `SourceEventIDs` is empty, we now fall back to `bullet.ID` as the dedup key. Added a direct unit test calling `deduplicateBullets` with two orphaned bullets (no SourceEventIDs, no company) to verify they are kept separate.
